libexec/daps-xmlwellformed

```diff
@@ -212,23 +131,112 @@ def parse_cli(args=None) -> argparse.Namespace:
     return args
 
 
-if __name__ == "__main__":
+def unique(lst: list):
+    """Makes a list unique (preserves order)
+
+    :param lst: the list that needs to be unique
+    :return: the (maybe) reduced list with unique elements
+    """
+    seen = {}
+
+    for item in lst:
+        if item in seen: continue
+        seen[item] = 1
+        yield item
+
+
+def get_all_files(xmlfile: str, xinclude: bool):
+    """Returns a list of all files when enabling XInclude processing
+
+    :param xmlfile: filename of XML file
+    :param xinclude: Enable or disable XInclude processing
+    :return:
+    """
+    ns = etree.FunctionNamespace(SUSE_NS)
+    ns['exists'] = exists
+    ns['abspath'] = abspath
+
+    xifiles = []
+    notfound = []
+    problems = []
+    error = None
+
+    try:
+        tree = etree.parse(xmlfile, parser=XMLPARSER)
+        transform = etree.XSLT(XSLT)
+        if xinclude:
+            result = transform(tree)
+            for entry in transform.error_log:
+                msg = entry.message
+                if msg.startswith("WARN:"):
```

Comment:
* For clarity, wouldn't it make more sense to use "NOTFOUND:" instead of "WARN:"? You literally take all "WARN" messages and add them to the documents that were not found
* If one of the files in my project is called "WARN:.xml" and it has some sort of XML issue, those errors might be swept under the table [?]. (By way of explanation: the error messages from libxml have the format "filename.xml:99:99: message" whereas you are using the incompatible format "MESSAGEWORD: message".)
  This may be a largely theoretical issue right now. One solution to it might be putting these "WARN:" messages on stdout instead of stderr (but that would preclude you from constructing a document there). Alternatively, if you have a way to add a "MESSAGEWORD: ..." before all output from libxml, that would solve the issue too but I am guessing that does not exist.

Reply:
> * For clarity, wouldn't it make more sense to use "NOTFOUND:" instead of "WARN:"?

Well, yes, that's probably better. Fixed that.

> * If one of the files in my project is called "WARN:.xml" and it has some sort of XML issue, those errors might be swept under the table [?]

I've looked at the code again and found one problem. The `split()` function should split only one time. I fixed that.
Basically, if message would be defined as `msg="WARN:WARN:.xml"`, then `msg.split(":", 1)` would lead to `['', 'WARN:.xml']` which is what we want.

Reply:
> [...] whereas you are using the incompatible format "MESSAGEWORD: message".)

I need to look into this again, maybe I can improve consistency.

Reply:
I used the key word "NOTFOUND"